### PR TITLE
feat: hasPendingMappings

### LINKED ISSUE #597: feat(mcm): new method has_pending_mapping
Please add the following method to `search_client`

```java
function has_pending_mappings(retrieve_mappings: boolean = false, opts: request_options) 
return has_pending_mappings_response

struct has_pending_mappings_response {
	pending: boolean,
	clusters: map<string, string[]>
}
```

```java
{
  "method": "GET",
  "url": "/1/clusters/mapping/pending?getClusters={retrieve_mappings}",
  "hostType": "read"
}
```

### PR DESCRIPTION
**Pull request by chloelbn**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Related Issue     | Fix #597
| Need Doc update   | yes

## Describe your change
- adds `hasPendingMappings` method

### CODE REVIEW

File: src/SearchClient.php [hunk 1]
```diff
@@ -380,4 +380,21 @@ public static function getSecuredApiKeyRemainingValidity($securedAPIKey)
 
         return $validUntil - time();
     }
+
+    /**
+     * @param boolean $retrieveMappings
+     * @param array $requestOptions
+     * @return array<string, boolean|array>
```
Comment (nunomaduro) [inline src/SearchClient.php]: We can get an array out of this?

File: src/SearchClient.php [hunk 2]
```diff
@@ -380,4 +380,21 @@ public static function getSecuredApiKeyRemainingValidity($securedAPIKey)
...
+     */
+    public function hasPendingMappings($retrieveMappings, $requestOptions = array())
```
Comment (nunomaduro) [inline src/SearchClient.php]: Tests are missing here, you are waiting for the CTS specs on this?

File: src/SearchClient.php (hunk 1 above)
Comment (chloelbn) [reply src/SearchClient.php]: the response would look like 
```
[
  'pending' => true,
  'clusters' => [
    'foo-test' => ["bar-0000-00-00"]
  ]
]
```

File: src/SearchClient.php [hunk 3]
```diff
@@ -380,4 +380,22 @@ public static function getSecuredApiKeyRemainingValidity($securedAPIKey)
...
         return $validUntil - time();
     }
+
+    /**
+     * @param bool  $retrieveMappings
+     * @param array $requestOptions
+     *
+     * @return array<string, boolean|array>
+     */
+    public function hasPendingMappings($retrieveMappings = false, $requestOptions = array())
```
Comment (julienbourdeau) [inline src/SearchClient.php]: I don't know if the name is already decided but usually when I see `hasSomething` method, I expect to get a boolean, not a "complex" object.
Maybe `getPendingMappings` could be a good name ¯\_(ツ)_/¯

Comment (chloelbn) [reply src/SearchClient.php]: it was indeed discussed that `getPendingMappings` would be a bit redundant with the parameter `retrieveMappings true|false`, and that `hasPendingMappings` would better convey what we're trying to achieve. Feel free to ping me if you want more information!

Comment (nunomaduro) [reply src/SearchClient.php]: As discussed, you need add `retrieveMappings` as part of request options. Just like : https://github.com/algolia/algoliasearch-client-python/pull/463/files.

**Pull request merged**